tests/python/test_multidevice.py

```diff
@@ -167,20 +167,28 @@ def multidevice_schedule(self) -> None:
     utils.is_pre_ampere(),
     reason="Flash Attention is only supported on Ampere and newer devices.",
 )
+@pytest.mark.parametrize("memory_format", ["bhse", "bshe"])
 @pytest.mark.mpi
-def test_sdpa(mpi_test):
+def test_sdpa(mpi_test, memory_format: str):
     d, b, s, h, e = mpi_test.size, 2, 1024, 12, 768
 
     if h % d != 0:
         pytest.skip(f"We only support even split, so {h} has to be divisible by {d}.")
 
     class Model(FusionDefinition):
+        def __init__(self, memory_format: str):
+            super().__init__()
+            self._memory_format = memory_format
+
         def definition(self) -> None:
             self.q, self.k, self.v, self.out_grad = [
                 self.define_tensor(
                     shape=[d, b, h // d, s, e // h],
                     contiguity=True,
                     dtype=DataType.BFloat16,
+                    stride_order=[4, 3, 2, 1, 0]
+                    if self._memory_format == "bhse"
+                    else [4, 3, 1, 2, 0],
```

Comment:
It was my intention to avoid using boolean here. I understood there are only two possibilities at this moment, but https://www.teamten.com/lawrence/programming/prefer-enums-over-booleans.html has stated reasons to prefer enum. (Well, I'm using str which is less typesafe than an enum. I'm happy to change that if you like)

> Since memory_format carries a special meaning in pytorch

Indeed: https://pytorch.org/docs/stable/tensor_attributes.html#torch-memory-format. How about `qkv_format` as [used in TransformerEngine](https://github.com/NVIDIA/TransformerEngine/blob/237b4930be8558a3ac97d34fb7e22f813216680c/transformer_engine/common/include/transformer_engine/fused_attn.h#L67)? 

> bhse/bshe reads confusing

Well, it's not that different from nhwc vs nchw. I'm open to better naming though!

Reply:
got'ya. Agree that it's probably easier to keep it somewhat future-proof.

In that case:
1. let's keep the intention for future expansion clear.
```
if qkv_format == "bhse":
elif qkv_format == "bshe":
else:
  raise NotImplementedError(f"unrecognized {qkv_format=}")
```
2. `qkv_format` sounds good enough to me. `bhse/bshe` under `qkv_format` also sounds good.